### PR TITLE
fix(TextField):  Remove caret from regex to correctly detect if the browser is Edge

### LINKED ISSUE #24333: Double eye icon on password input box
### Environment Information

- **Package version(s)**: (fill this out)
- **Browser and OS versions**: (fill this out if relevant)

### Describe the issue:

when a user tries to input a password, it shows double eye icon 

image

#### Actual behavior:

image

#### Expected behavior:

there should be only one eye icon instead of two

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Blocking feature deployment

Products/sites affected: Microsoft security

### PR DESCRIPTION
**Pull request by sopranopillow**
## Current Behavior

image

## New Behavior

image

## Related Issue(s)

Fixes #24333

Related #23482

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/27a3480f5361aa5d99f8978817d8395d9037e3b7'>27a3480f5361aa5d99f8978817d8395d9037e3b7</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/BaseButton-mount_base.html">759</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/BaseButton-mount.html">826</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Breadcrumb-mount_base.html">2300</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Breadcrumb-mount.html">2274</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Checkbox-mount_base.html">2274</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Checkbox-mount.html">2276</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/CheckboxBase-mount_base.html">2000</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/CheckboxBase-mount.html">1984</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ChoiceGroup-mount_base.html">3950</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ChoiceGroup-mount.html">3927</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ComboBox-mount_base.html">798</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ComboBox-mount.html">777</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/CommandBar-mount_base.html">9093</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/CommandBar-mount.html">8969</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ContextualMenu-mount_base.html">9751</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ContextualMenu-mount.html">9682</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/DefaultButton-mount_base.html">987</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/DefaultButton-mount.html">952</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/DetailsRow-mount_base.html">3039</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/DetailsRow-mount.html">3066</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/DetailsRowFast-mount_base.html">3067</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/DetailsRowFast-mount.html">3096</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/DetailsRowNoStyles-mount_base.html">2933</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/DetailsRowNoStyles-mount.html">2895</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Dialog-mount_base.html">2324</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Dialog-mount.html">2280</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/DocumentCardTitle-mount_base.html">154</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/DocumentCardTitle-mount.html">152</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Dropdown-mount_base.html">2786</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Dropdown-mount.html">2792</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/FocusTrapZone-mount_base.html">1558</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/FocusTrapZone-mount.html">1553</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/FocusZone-mount_base.html">1530</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/FocusZone-mount.html">1514</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/IconButton-mount_base.html">1421</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/IconButton-mount.html">1397</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Label-mount_base.html">330</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Label-mount.html">327</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Layer-mount_base.html">2737</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Layer-mount.html">2719</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Link-mount_base.html">450</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Link-mount.html">430</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/MenuButton-mount_base.html">1243</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/MenuButton-mount.html">1223</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/MessageBar-mount_base.html">2011</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/MessageBar-mount.html">1867</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Nav-mount_base.html">2731</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Nav-mount.html">2653</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/OverflowSet-mount_base.html">1025</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/OverflowSet-mount.html">978</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Panel-mount_base.html">1884</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Panel-mount.html">1807</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Persona-mount_base.html">864</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Persona-mount.html">837</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Pivot-mount_base.html">1164</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Pivot-mount.html">1146</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/PrimaryButton-mount_base.html">1094</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/PrimaryButton-mount.html">1098</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Rating-mount_base.html">6699</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Rating-mount.html">6650</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/SearchBox-mount_base.html">1143</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/SearchBox-mount.html">1109</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Shimmer-mount_base.html">2252</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Shimmer-mount.html">2230</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Slider-mount_base.html">1686</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Slider-mount.html">1691</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/SpinButton-mount_base.html">4078</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/SpinButton-mount.html">3972</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Spinner-mount_base.html">411</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Spinner-mount.html">404</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/SplitButton-mount_base.html">2503</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/SplitButton-mount.html">2479</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Stack-mount_base.html">485</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Stack-mount.html">483</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/StackWithIntrinsicChildren-mount_base.html">1869</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/StackWithIntrinsicChildren-mount.html">1874</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/StackWithTextChildren-mount_base.html">4696</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/StackWithTextChildren-mount.html">4641</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/SwatchColorPicker-mount_base.html">9204</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/SwatchColorPicker-mount.html">9023</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/TagPicker-mount_base.html">2081</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/TagPicker-mount.html">2136</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/TeachingBubble-mount_base.html">76656</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/TeachingBubble-mount.html">75486</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Text-mount_base.html">409</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Text-mount.html">413</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/TextField-mount_base.html">1155</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/TextField-mount.html">1140</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ThemeProvider-mount_base.html">980</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ThemeProvider-mount.html">978</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ThemeProvider-virtual-rerender_base.html">643</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ThemeProvider-virtual-rerender.html">630</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1429</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1405</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Toggle-mount_base.html">731</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/Toggle-mount.html">731</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/buttonNative-mount_base.html">107</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24337/perf-test/buttonNative-mount.html">135</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/TextField/TextField.types.ts
Comment (smhigley) [inline packages/react/src/components/TextField/TextField.types.ts:287]: I'd lean towards renaming this to be clearer -- "built-in" could refer to either a button built into Fluent, or built into the browser (I actually initially thought it meant the opposite of what it does 😅). Maybe something like "enableBrowserPasswordReveal"?

**Review by smhigley — CHANGES_REQUESTED** [review]
I think it'd be good to hold off on merging this until we make a final decision on whether the browser button should be the default vs. the Fluent button.

Other than that + the prop name comment, the rest looks good to me!

File: packages/react/src/components/TextField/TextField.types.ts
Comment (spmonahan) [inline packages/react/src/components/TextField/TextField.types.ts:291]: This comment lead me to dig deeper into this issue and [this PR](https://github.com/microsoft/fluentui/commit/3a9fd36f3096ca19c100c956dcddb4196cf9ef08#diff-abaa20fc65c181b1cd4b9ae6857c16ff4cd6f7ee6f2a8d7ed96c50e9291610ba) seems to have broken the behavior.

Fundamentally the bug is that [we're sniffing the user agent](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/TextField/TextField.base.tsx#L675) rather than feature detecting but changing the regex will "fix" the behavior and only show a single reveal button (change `^Edg` to `Edg`).

Off the top of my head I see two things we can do here:

1. Change this PR to fix the bug and remove the new prop, essentially restoring the old behavior.
2. Keep the new prop and behavior from this PR and remove the old UA sniffing.

**Review by spmonahan — CHANGES_REQUESTED** [review]

File: packages/react/src/components/TextField/TextField.types.ts
Comment (spmonahan) [reply packages/react/src/components/TextField/TextField.types.ts:291]: I should add that I'm not sure it's possible to feature detect the password reveal button so UA sniffing may be the best we can do.

**Review by spmonahan — APPROVED** [review]

**Review by smhigley — APPROVED** [review]

**Pull request merged**
